cli/src/cli/keygen.rs

```diff
@@ -0,0 +1,50 @@
+use error::DefaultResult;
+use holochain_dpki::{
+    bundle::KeyBundle,
+    keypair::{Keypair, SEEDSIZE},
+    util::PwHashConfig,
+};
+use holochain_sodium::{pwhash, random::random_secbuf, secbuf::SecBuf};
+use rpassword;
+use std::{
+    fs::{create_dir_all, File},
+    io::prelude::*,
+    path::PathBuf,
+};
+
+pub fn keygen() -> DefaultResult<()> {
+    let passphrase = rpassword::read_password_from_tty(Some("Passphrase: ")).unwrap();
+
+    let mut seed = SecBuf::with_secure(SEEDSIZE);
+    random_secbuf(&mut seed);
+    let mut keypair = Keypair::new_from_seed(&mut seed).unwrap();
+    let passphrase_bytes = passphrase.as_bytes();
+    let mut passphrase_buf = SecBuf::with_insecure(passphrase_bytes.len());
+    passphrase_buf
+        .write(0, passphrase_bytes)
+        .expect("SecBuf must be writeable");
+
+    let bundle: KeyBundle = keypair
+        .get_bundle(
+            &mut passphrase_buf,
+            "hint".to_string(),
+            Some(PwHashConfig(
+                pwhash::OPSLIMIT_INTERACTIVE,
+                pwhash::MEMLIMIT_INTERACTIVE,
+                pwhash::ALG_ARGON2ID13,
+            )),
+        )
+        .unwrap();
+
+    let path = match directories::UserDirs::new() {
+        Some(user_dirs) => user_dirs.home_dir().join(".holochain").join("keys"),
```

Comment:
Based on what I added into the tree about directories, I propose that we create a new crate, or shareable file, that keeps track of and documents all the “magic strings” representing paths on devices. I find that they get scattered in the code and the act as high risk points of failure for our system I think. If we could start in that direction in this Pr that would be amazing

Reply:
Also I'm in favor of creating these directories with the XDG_CONFIG convention in mind. The `directories` crate already does that for you, we just have to use it.